code/HabitTracker/app/src/main/java/com/example/habittracker/activities/HabitViewActivity.java

```diff
@@ -56,16 +140,34 @@ public void onClick(View view) {
             public void onClick(View view) {
                 Intent intent = new Intent(getApplicationContext(),ProgressTrackingActivity.class);
                 Bundle bundle = new Bundle();
-                Habit habit = getHabit();
                 bundle.putSerializable("habit", habit);
```

Comment:
Probably because you had to make habit final in order to pass it in there.
But yes, I believe it passes the habit object properly (I had a different intent/content thing going on), a little bit above this line